src/session.ts

```diff
@@ -87,6 +103,22 @@ async function updateSession(request: NextRequest, debug: boolean) {
   }
 }
 
+function parseRoute(route: string): string {
```

Comment:
I find the naming confusing here, it's not really parsing a route, it's taking a global-like path expression and returning a regex string to test if a path matches it.

Could we rename it to something more meaningful/specific and also make it return the regexp itself:

```ts
function getMiddlewareAuthPathRegex(pathGlob: string) {
  …
  regex = new RegExp(tokensToRegexp(tokens).source);
  …
}
```